paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/TransformSpecToElement.kt

```diff
@@ -47,8 +47,8 @@ internal class TransformSpecToElement {
                 spec.field as SectionFieldSpec.Name,
                 focusRequesterCount
             )
-            SectionFieldSpec.Country -> transform(
-                spec.field as SectionFieldSpec.Country
+            is SectionFieldSpec.Country -> transform(
```

Comment:
If we define the `IdentifierSpec` as sealed class and `object`s, then we can use `when (spec.field.identifier)` here and avoid the type check. (which I think is more appropriate because it's an **identifier**)

Reply:
IdentifierSpec is something that will be coming from the server, so we can't really seal it, it will just be json

Reply:
We'll still use a json parser to map it to a `IdentifierSpec` type instead of passing it as `JSONObject` right? If you look at [PaymentIntentJsonParser](https://github.com/stripe/stripe-android/blob/79386291cfde4e091abac8917fcf277de4c3ee9e/payments-core/src/main/java/com/stripe/android/model/parsers/PaymentIntentJsonParser.kt#L10), it's returning a `PaymentIntent` from server response, which is a subclass of sealed [StripeIntent](https://github.com/stripe/stripe-android/blob/c294238538f0a6a1601228c263a78bf7b110adb8/payments-core/src/main/java/com/stripe/android/model/StripeIntent.kt#L12)

Reply:
Another more similar example is [NextActionData](https://github.com/stripe/stripe-android/blob/c294238538f0a6a1601228c263a78bf7b110adb8/payments-core/src/main/java/com/stripe/android/model/StripeIntent.kt#L143), it's parsed from different parsers(e.g [RedirectToUrlParser](https://github.com/stripe/stripe-android/blob/ad426ca33105e7bc5f17c027f8146354d13f75ae/payments-core/src/main/java/com/stripe/android/model/parsers/NextActionDataParser.kt#L48)), since they all implement the same sealed interface, we were able to save them in a [map](https://github.com/stripe/stripe-android/blob/55d9b07317176291e45090f5d4ad8fe5ee18937a/payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt#L27) in a registry for look up